### DISCUSSION TITLE
Running Volar as a standalone language server?

### DISCUSSION THREAD

**Discussion by maxbrieiev** [vuejs/language-tools#338] [Q&A] [answered]
Is it possible to run volar as a standalone LSP server?

What I want is to run volar (LSP server) as a separate process, and connect to it by an LSP client (different than VSCode). That is, I want to use volar with a different editor.

Where should I start? There is a package [vscode-vue-languageservice](https://www.npmjs.com/package/vscode-vue-languageservice), but I don't get the way to use it. It doesn't provide any executables. Should I write some wrapper script to start LSP server and accept some connection details from command line, like `--stdio` or `--host`, `--port`?

Thanks.

**Comment by johnsoncodehk** [comment, marked as answer]
I'm studying on it, maybe you can get some hints from here:

- https://github.com/yaegassy/coc-volar-experimental
- https://github.com/yaegassy/coc-volar-experimental/issues/2

**Reply by maxbrieiev** [reply]
Thank you. I will look into it.

**Comment by fenixjiang** [comment]
Does it mean that I can directly run `node server.js --stdio or --socket=4711` to start the server and use any LSP client to connect to it?

I tried that, but it failed. The server exited directly when the client connected, or the server failed to start--started in the socket way.
